tests/lax_numpy_test.py

```diff
@@ -5857,6 +5866,87 @@ def testUfuncInputTypes(self, name, arg_dtypes):
+        ('log1p', 'pos'), ('log1p', 'neg'), ('log1p', 'posj'), ('log1p', 'negj'),
+        ('log1p', 'ninf'), ('log1p', 'ninfj'), ('log1p', 'pinfj'),
+        ('log10', 'q1'), ('log10', 'q2'), ('log10', 'q3'), ('log10', 'q4'), ('log10', 'zero'),
+        ('log10', 'ninf'), ('log10', 'ninfj'), ('log10', 'pinf'), ('log10', 'pinfj'),
+        ('sinh', 'q1'), ('sinh', 'q2'), ('sinh', 'q3'), ('sinh', 'q4'),
+        ('sinh', 'pos'), ('sinh', 'neg'), ('sinh', 'ninf'), ('sinh', 'pinf'),
+        ('cosh', 'pos'), ('cosh', 'neg'), ('cosh', 'ninf'), ('cosh', 'pinf'),
+        ('cosh', 'q1'), ('cosh', 'q2'), ('cosh', 'q3'), ('cosh', 'q4'),
+        ('tan', 'q1'), ('tan', 'q2'), ('tan', 'q3'), ('tan', 'q4'),
+        ('tan', 'negj'), ('tan', 'posj'), ('tan', 'ninf'), ('tan', 'ninfj'), ('tan', 'pinf'), ('tan', 'pinfj'),
+        # warning: numpy square is incorrect on inputs with large absolute value
+        ('square', 'q1'), ('square', 'q2'), ('square', 'q3'), ('square', 'q4'),
+        ('square', 'ninfj'), ('square', 'pinfj'),
+        ('sinc', 'q1'), ('sinc', 'q2'), ('sinc', 'q3'), ('sinc', 'q4'),
+        ('sign', 'q1'), ('sign', 'q2'), ('sign', 'q3'), ('sign', 'q4'),
+        ('sign', 'negj'), ('sign', 'posj'), ('sign', 'ninf'), ('sign', 'ninfj'), ('sign', 'pinf'), ('sign', 'pinfj'),
+    } or (
+      name in {'arcsin', 'arccos', 'arctan', 'arcsinh', 'arccosh', 'arctanh'}
+    ) or (
+      jtu.test_device_matches(["cpu"]) and (name, region) in {
+        ('exp2', 'q1'), ('exp2', 'q4'),
+        ('sinh', 'negj'), ('sinh', 'posj'),
+      }
+    ) or (
+      jtu.test_device_matches(["cpu"]) and dtype == np.complex64 and (name, region) in {
+        ('square', 'negj'), ('square', 'neg'), ('square', 'posj'), ('square', 'pos'),
+      }
+    ):
+      self.skipTest(f"np.{name} & jnp.{name} have differences in the given region.")
+    jnp_op = getattr(jnp, name)
+    np_op = getattr(np, name)
+    rtol = {np.complex64: 1e-6, np.complex128: 1e-14}[dtype]
+
+    with warnings.catch_warnings():
+      # Silence RuntimeWarning: overflow encountered in cast
+      warnings.simplefilter("ignore")
+
+      args = (jtu.complex_plane_sample(dtype=dtype, size_re=10),)
+      result = np.asarray(jnp_op(*args))
+      expected = np_op(*args)
+
+      s0, s1 = (result.shape[0] - 3) // 2, (result.shape[1] - 3) // 2
+
+      s = dict(
+        q1=(slice(s0 + 2, -1), slice(s1 + 2, -1)),
+        q2=(slice(s0 + 2, -1), slice(1, s1 + 1)),
+        q3=(slice(1, s0 + 1), slice(1, s1 + 1)),
+        q4=(slice(1, s0 + 1), slice(s1 + 2, -1)),
+        neg=(s0 + 1, slice(1, s1 + 1)),
+        pos=(s0 + 1, slice(s1 + 2, -1)),
+        negj=(slice(1, s0 + 1), s1 + 1),
+        posj=(slice(s0 + 2, -1), s1 + 1),
+        ninf=(slice(None), 0),
+        pinf=(slice(None), -1),
+        ninfj=(0, slice(None)),
+        pinfj=(-1, slice(None)),
+        zero=(s0 + 1, s1 + 1),
+      )[region]
+
+      np.testing.assert_allclose(result[s], expected[s], rtol=rtol, atol=0)
```

Comment:
Prefer `self.assertAllClose` within JAX tests: in particular it automatically adjusts tolerances based on the dtype, so you may not need to do so manually.

Reply:
Fixed.